mii/config.py

```diff
@@ -99,29 +153,124 @@ def meta_tensor_or_sys_mem(cls, values):
             )
         return values
 
-    class Config:
-        validate_all = True
-        validate_assignment = True
-        use_enum_values = True
-        extra = 'forbid'
-        json_encoders = {torch.dtype: lambda x: str(x)}
+    @root_validator
+    def zero_dtype_valid(cls, values):
+        if values.get("enable_zero"):
+            if values.get("ds_config").get("fp16", {}).get("enabled", False):
+                assert (
+                    values.get("dtype") == DtypeEnum.float16
+                ), "ZeRO FP16 enabled, `dtype` must be set to `torch.float16`"
+            else:
+                assert (
+                    values.get("dtype") == DtypeEnum.float32
+                ), "ZeRO FP16 disabled, `dtype` must be set to `torch.float32`"
+        return values
 
+    @root_validator
+    def deepspeed_or_zero(cls, values):
+        assert not (
+            values.get("enable_deepspeed") and values.get("enable_zero")
+        ), "DeepSpeed and ZeRO cannot both be enabled, select only one"
+        return values
 
-class ReplicaConfig(BaseModel):
-    hostname: str = ""
-    tensor_parallel_ports: List[int] = []
-    torch_dist_port: int = None
-    gpu_indices: List[int] = []
 
-    class Config:
-        validate_all = True
-        validate_assignment = True
+class MIIConfig(DeepSpeedConfigModel):
```

Comment:
TODO: Bring the comments on what each option does (from `deployment.py`) to here. Make compatible with the auto-generated pydantic docs (and create the docs for MII)